server/src/main/java/org/opensearch/indices/replication/SegmentReplicationSourceHandler.java

```diff
@@ -176,6 +187,15 @@ public synchronized void sendFiles(GetSegmentFilesRequest request, ActionListene
         }
     }
 
+    // Update target replication checkpoint on source for node-node communication. For remote store enabled indices, checkpoint
+    // update on source is performed via separate UPDATE_VISIBLE_CHECKPOINT transport call
+    private void updateVisibleCheckpointForShard(String allocationId, ReplicationCheckpoint replicationCheckpoint) {
```

Comment:
The is component is only invoked with node-node. So I think this is unnecessary?

Reply:
Updated